File: IMDB-CSharp/IMDB/Models/Film.cs
using System.ComponentModel.DataAnnotations;
using System.Web.Mvc;

namespace IMDB.Models
{
    public class Film
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MinLength(1)]
        public string Name { get; set; }

        [Required]
        [MinLength(1)]
        public string Genre { get; set; }

        [Required]
        [MinLength(1)]
        public string Director { get; set; }

        [Range(1900, 2100)]
        public int Year { get; set; }
    }
}

File: IMDB-JavaScript/models/Film.js
const mongoose = require('mongoose');

let filmSchema = mongoose.Schema({
    name: {type: 'string', required: 'true'},
    genre: {type: 'string', required: 'true'},
    director: {type: 'string', required: 'true'},
    year: {type: 'number', required: 'true'},
});

let Film = mongoose.model('Film', filmSchema);

module.exports = Film;